### DISCUSSION TITLE
Files access

### DISCUSSION THREAD

**Discussion by litozpl** [fat-tire/resolve#37] [Q&A] [answered]
Hello everyone,

Thank you so much for this technique! I was in a bad situation and you help me a lot, DA is working again ! :)
So now I have Davinci Resolve Studio 18.5B (Build 39) on my Ubuntu 23.04.

But with containers, Resolve only access to the directory "inside" the container. So its "mounts/resolve-home". I tried to mount a USB key and solid state. But, DA do not see files inside directory. My system can, but don't have permission when I mount in the mounts/resolve-home/XXX.

Some one have a clue or a solution ?

Thanks again for this :)

Ps: I post the same message on the Blackmagic forum

**Comment by fat-tire** [comment, marked as answer]
This is easy to do!

1.  Plug in the SSD or USB on your host computer.

It will hopefully mount properly, and you'll be able to access it from your computer.  If you have problems writing to the SSD, make sure that the ownership and permissions of the drive are set properly.  (Consider using `chown` or `chmod` as needed)

2.  Determine he path for the mounted SSD/USB.  One way to do this is with the `mount` command, which will show you that the SSD (in this example) is located at:

```bash
/dev/sda1 on /media/litozpl/mySSD type ext4 (rw,nosuid,nodev,relatime,errors=remount-ro,stripe=4,uhelper=udisks2)
```

This means that it is on the sda (serial device A, aka a plugged in USB), partition 1, and it's mounted at `/media/litozpl/mySSD`.  The other stuff in the parenthesis gives you details about how the device is mounted, that it is read/write, etc.)

You can also find the path information in Ubuntu by running the `gnome-disks` application (see more [here](https://manpages.ubuntu.com/manpages/focal/en/man1/gnome-disks.1.html)). Click the disk you want to mount, and look under the  **Contents** option where you'll see something like **Mounted at /media/litozpl/mySSD**.

3.  Now you can run `resolve.sh` like this:

```bash
RESOLVE_BIND_SOURCES=("/media/litozpl/mySSD") RESOLVE_BIND_TARGETS=("/mnt/mySSD") ./resolve.sh
```     
And you should be able to access the files from within the container (and from DaVinci Resolve) at `/mnt/mySSD`.

Alternatively, if you'll be mounting this drive every time, you can create a file called `resolve.rc` containing:

```bash
RESOLVE_BIND_SOURCES=("/media/litozpl/mySSD")
RESOLVE_BIND_TARGETS=("/mnt/mySSD")
```

Add any other `RESOLVE_` variables or special BASH instructions you like, then:

```bash
RESOLVE_RC_PATH=./resolve.rc ./resolve.sh
```

You can add additional SOURCES/TARGETS pairs if you wanted to mount more than one SSD/USB drive.

See the `README.md` file for more ideas for things to put in the `resolve.rc`.  Remember that the contents of resolve.rc will be executed when you start resolve this way, so consider any security implications -- for example-- don't make that file writable to other untrusted users.)

Hope this is useful info. Enjoy!

**Comment by litozpl** [comment]
Thank you so much Fat-Tire !

Every thing's fine. Now, I can start my jobs :)
